x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java

```diff
@@ -185,18 +258,30 @@ public static TaskParams fromXContent(XContentParser parser) {
 
         private final String modelId;
         private final long modelBytes;
+        private final Integer inferenceThreads;
+        private final Integer modelThreads;
 
-        public TaskParams(String modelId, long modelBytes) {
+        public TaskParams(String modelId, long modelBytes, @Nullable Integer inferenceThreads, @Nullable Integer modelThreads) {
             this.modelId = Objects.requireNonNull(modelId);
             this.modelBytes = modelBytes;
             if (modelBytes < 0) {
                 throw new IllegalArgumentException("modelBytes must be non-negative");
             }
+            this.inferenceThreads = inferenceThreads;
+            if (inferenceThreads != null && inferenceThreads < 1) {
+                throw new IllegalArgumentException(INFERENCE_THREADS + " must be positive");
+            }
+            this.modelThreads = modelThreads;
+            if (modelThreads != null && modelThreads < 1) {
+                throw new IllegalArgumentException(MODEL_THREADS + " must be positive");
+            }
```

Comment:
What do you think of populating these with defaults here? Its nice to be super clear to the user that we are defaulting to 1.